xarray/namedarray/core.py

```diff
@@ -703,6 +718,154 @@ def sizes(self) -> dict[_Dim, _IntOrUnknown]:
         """Ordered mapping from dimension names to lengths."""
         return dict(zip(self.dims, self.shape))
 
+    def chunk(
+        self,
+        chunks: (
+            int
+            | Literal["auto"]
+            | tuple[int, ...]
+            | tuple[tuple[int, ...], ...]
+            | Mapping[Any, None | int | tuple[int, ...]]
+        ) = {},
```

Comment:
A lot of typing here. Do we like them all?

Reply:
Ccing @TomNicholas

Reply:
I think given https://github.com/pydata/xarray/pull/8341 we can remove the two `tuple` types. However as the deprecation cycle for passing a tuple of chunks to eg. `DataArray.chunk` has only just begun, I think we will need to add a piece of code to change tuples to mappings of chunks in the same place as the deprecation warning is raised in #8341. (cc @max-sixty ) We still need the int, Literal, and Mapping.

Reply:
The deprecation cycle is for DataArray/Variable though. So I think we should try to keep NamedArray as close to the ideal as possible, and solve the backwards compat issues in DataArray/Variable.

I think we should be careful inheriting api that we aren't satisfied with, it's easy to forget once it's merged.

Reply:
> I think we should be careful inheriting api that we aren't satisfied with, it's easy to forget once it's merged.

That's fair enough.

> The deprecation cycle is for DataArray/Variable though.

I think we agree - I'm just trying to make it clear that having these arguments in DataArray/Variable but not in NamedArray requires some compatibility code that hasn't been written yet, at least until the deprecation cycle is finished.

Reply:
I don't think this is actually resolved? Unless it's being deferred to a later PR?